core/lib/Cardano/Address/Script.hs

```diff
@@ -266,35 +301,75 @@ foldScript fn zero = \case
 -- | Validate a 'Script', semantically
 --
 -- @since 3.0.0
-validateScript :: Script -> Either ErrValidateScript ()
-validateScript = \case
-    RequireSignatureOf (KeyHash bytes) -> do
-        when (BS.length bytes /= credentialHashSize) $ Left WrongKeyHash
+validateScript :: Script KeyHash -> Either ErrValidateScript ()
+validateScript script = do
+    let validateKeyHash (KeyHash bytes) =
+            when (BS.length bytes /= credentialHashSize) $ Left WrongKeyHash
+    validateScript' validateKeyHash script
+
+validateScript'
+    :: Eq elem
+    => (elem -> Either ErrValidateScript ())
+    -> Script elem
+    -> Either ErrValidateScript ()
+validateScript' validateRequireSignatureOf = \case
```

Comment:
I generally try to avoid `'` as much as possible when naming functions as there are often better alternative. Something like: `validateAbstractScript` or `validateScriptWith` for instance.

Reply:
done, see state as in https://github.com/input-output-hk/cardano-addresses/pull/99/commits/070e6bf1e09840eb7f88e782baed7eeee56dcabe